### PR TITLE
Implement edit user endpoint

### PR DESCRIPTION
**Pull request by Anurav-Garg**

### CODE REVIEW

File: src/controllers/user/editUser.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }),
+});
+
+export const editUserValidator = validate(dataSchema);
+
+export const editUser = async (req: Request, res: Response) => {
+  try {
+    let author: User | null = null;
+
+    // this try-catch block will be removed when auth is added
```
Comment (soumitradev) [inline src/controllers/user/editUser.ts:43]: Remove this now, no need to wait for OAuth

File: src/controllers/user/editUser.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    const degrees: DegreeList = req.body.degrees;
+
+    if (degrees.length === 2 && !isAValidDegreeCombination(degrees)) {
+      return res.status(400).json({
+        message: "Dualites may only have one BE degree and one MSc degee",
+      });
+    }
+
+    try {
+      await userRepository.manager.transaction(
```
Comment (soumitradev) [inline src/controllers/user/editUser.ts:69]: Do you need to access the manager class here, or is `userRepository.transaction` sufficient?

File: src/controllers/user/editUser.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            .set({ degrees })
+            .where("email = :email", { email: req.body.email })
+            .execute();
+
+          // Update degrees for user's timetables
+          await transactionalEntityManager
+            .createQueryBuilder()
+            .update("timetable")
+            .set({ degrees })
+            .where("author = :author", { author: author?.id })
```
Comment (soumitradev) [inline src/controllers/user/editUser.ts:84]: Here if possible fetch the ID of the user being updated using the query above this

**Review by soumitradev — CHANGES_REQUESTED** [review]

**Comment by soumitradev** [comment]
Also from next time I'd recommend making the endpoint, and the router changes as separate commits

File: src/controllers/user/editUser.ts (hunk 2 above)
Comment (Anurav-Garg) [reply src/controllers/user/editUser.ts:69]: From what I saw in the docs, only the data source and the manager object have the transaction method

File: src/controllers/user/editUser.ts [hunk 4]
```diff
@@ -69,19 +50,23 @@ export const editUser = async (req: Request, res: Response) => {
...
+          const authorId = await transactionalEntityManager
             .createQueryBuilder()
             .update("user")
             .set({ degrees })
             .where("email = :email", { email: req.body.email })
-            .execute();
+            .returning("*")
+            .execute()
+            .then((response) => {
+              return response.raw[0].id;
```
Comment (soumitradev) [inline src/controllers/user/editUser.ts:61]: Grab this from identifiers, not raw.

**Review by soumitradev — CHANGES_REQUESTED** [review]

**Review by soumitradev — APPROVED** [review]
LGTM

**Pull request merged**